libs/src/services/discoveryProvider/index.js

```diff
@@ -43,17 +44,21 @@ class DiscoveryProvider {
         whitelistMap[urlJoin(url, '/version')] = url
       })
 
-      let resp = await Utils.raceRequests(Object.keys(whitelistMap), (url) => {
-        pick = whitelistMap[url]
-      }, {})
-
-      const isValid = pick && resp.data.service && (resp.data.service === serviceType.DISCOVERY_PROVIDER)
-      if (isValid) {
-        console.info('Initial discovery provider was valid')
-        endpoint = pick
-      } else {
-        console.info('Initial discovery provider was invalid, searching for a new one')
-        endpoint = await this.ethContracts.selectDiscoveryProvider(this.whitelist)
+      try {
+        let resp = await Utils.raceRequests(Object.keys(whitelistMap), (url) => {
+          pick = whitelistMap[url]
+        }, {})
+  
+        isValid = pick && resp.data.service && (resp.data.service === serviceType.DISCOVERY_PROVIDER)
+        if (isValid) {
+          console.info('Initial discovery provider was valid')
+          endpoint = pick
+        } else {
+          console.info('Initial discovery provider was invalid, searching for a new one')
+          endpoint = await this.ethContracts.selectDiscoveryProvider(this.whitelist)
+        }
+      } catch(e) {
+        console.error('Could not select a discprov from the whitelist', e)
```

Comment:
so this looks good to me for local testing - but im not sure what the implications are for production dapp initialization. it might be preferable in that case to actually throw out...maybe we check the length of the whitelist and conditionally throw?

Reply:
@hareeshnagaraj so i ran this against staging with the staging whitelist and it works. Basically if there is a valid service up that it can select, it doesn't get to this code branch